test/test_workertype.js

```diff
@@ -0,0 +1,519 @@
+'use strict';
+var base = require('taskcluster-base');
+var data = require('../provisioner/data.js');
+var _ = require('lodash');
+var slugid = require('slugid');
+
+var cfg = base.config({
+  defaults: require('../config/defaults'),
+  profile: require('../config/test'),
+  envs: [
+    'provisioner_publishMetaData',
+    'provisioner_awsInstancePubkey',
+    'provisioner_awsKeyPrefix',
+    'azure_accountName',
+    'azure_accountKey',
+  ],
+  filename: 'taskcluster-aws-provisioner',
+});
+
+var keyPrefix = cfg.get('provisioner:awsKeyPrefix');
+//var pubKey = cfg.get('provisioner:awsInstancePubkey');
+var provisionerId = cfg.get('provisioner:id');
+//var maxInstanceLife = cfg.get('provisioner:maxInstanceLife');
+
+// We don't use influx yet, but may as well include it
+/* var influx = new base.stats.Influx({
```

Comment:
For testing you can also use the `base.stats.NullDrain`. it has a method `pendingPoints` so you can get a count of how many pending properties there is...